File: Yandex/Practicum/Algorightms_and_Data_Structures/1_sprint/A.Значение_функции/solution.py
import sys


def solution(a: int, x: int, b: int, c: int) -> int:
    """
    Time Complexity O(1)
    Memory Complexity O(1)
    """
    return a * x**2 + b * x + c


if __name__ == '__main__':
    a, x, b, c = map(int, sys.stdin.readline().rstrip().split())
    sys.stdout.write(str(solution(a, x, b, c)))

File: Yandex/Practicum/Algorightms_and_Data_Structures/1_sprint/A.Значение_функции/solution.cpp
#include <iostream>
#include <cmath>


int main() {
    /*
    Time Complexity O(1)
    Memory Complexity O(1)
    */
    int a, x, b, c;
    std::cin >> a >> x >> b >> c;
    int result = a * std::pow(x, 2) + b * x + c;
    std::cout << result << std::endl;
}